src/core/imap/MCIMAPSession.cpp

```diff
@@ -1468,11 +1468,22 @@ Array * /* IMAPFolder */ IMAPSession::fetchAllFolders(ErrorCode * pError)
         }
 
         if (!hasInbox) {
-            r = mailimap_list(mImap, "", "INBOX", &imap_folders);
-            Array * inboxResult = resultsWithError(r, imap_folders, pError);
-            if (* pError == ErrorConnection || * pError == ErrorParse)
-                mShouldDisconnect = true;
-            result->addObjectsFromArray(inboxResult);
+            mc_foreacharray(IMAPFolder, folder, result) {
+                if (folder->flags() & IMAPFolderFlagInbox) {
+                    // some mail providers use non-standart name for inbox folder
+                    hasInbox = true;
+                    folder->setPath(MCSTR("INBOX"));
+                    break;
+                }
+            }
+
+            if (!hasInbox) {
+                r = mailimap_list(mImap, "", "INBOX", &imap_folders);
+                Array * inboxResult = resultsWithError(r, imap_folders, pError);
+                if (* pError == ErrorConnection || * pError == ErrorParse)
+                    mShouldDisconnect = true;
+                result->addObjectsFromArray(inboxResult);
```

Comment:
Could you add an extra `hasInbox = true;` just to make it consistent?
Could you let me know which server has this issue?
Thanks!